transhumandesign/kag-base
# Add chicken emote; Allow players to sit on eggs to incubate them

## Status

- **READY**: this PR is (to the best of your knowledge) ready to be incorporated into the game.

## Description

![image](https://user-images.githubusercontent.com/6610341/143255701-f8c96189-a918-4a89-810c-116855ed7d4d.png)

This PR adds a new chicken emote (using one of the existing chicken sprite frames). It also adds the ability for players to crouch on eggs to incubate them. An incubated egg will hatch 3x faster than a regular egg. Whilst crouching on the egg you will automatically do the chicken emote. Your emote will be cleared if you leave.

## Steps to Test or Reproduce

I've tested this locally in sandbox.

- Open sandbox and spawn an egg
- Move over it as knight/builder/archer and crouch
- You should start incubating the egg. Your character will do the chicken emote, and the egg should hatch 3x faster than an un-incubated egg.
